src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/LPG2RDFConfigurationReader.java

```diff
@@ -0,0 +1,107 @@
+
+        if (!lpg2rdfConfigs.hasNext()){
+            throw new IllegalArgumentException("LPGtoRDFConfiguration is required!");
+        }
+        final Resource lpg2rdfConfig = lpg2rdfConfigs.next();
+        if(lpg2rdfConfigs.hasNext()){
+            throw new IllegalArgumentException("More than one instance of LPGtoRDFConfiguration!");
+        }
+
+        final String validDataType = lpg2Rdf.getNsPrefixURI("xsd").concat("anyURI");
+
+        final Node label = getLabelPredicate(lpg2rdfConfig, validDataType);
+        final NodeMapping nodeMapping = getNodeMapping(lpg2Rdf, lpg2rdfConfig);
+
+
+        return new LPG2RDFConfigurationImpl(label, nodeMapping);
+    }
+
+    public Node getLabelPredicate(final Resource lpg2rdfConfig, final String validDataType){
+        if(!lpg2rdfConfig.hasProperty(LPG2RDF.labelPredicate)){
+            throw new IllegalArgumentException("labelPredicate is required!");
+        }
+
+        final StmtIterator labelPredicateIterator = lpg2rdfConfig.listProperties(LPG2RDF.labelPredicate);
+
+        if(!labelPredicateIterator.hasNext()){
+            throw new IllegalArgumentException("labelPredicate is required!");
+        }
+        final RDFNode labelPredicateObj = labelPredicateIterator.next().getObject();
+        if(labelPredicateIterator.hasNext()){
+            throw new IllegalArgumentException("More than one labelPredicate!");
+        }
+        if (!labelPredicateObj.asLiteral().getDatatypeURI().equals(validDataType)){
+            throw new IllegalArgumentException("labelPredicate is invalid!");
+        }
+        final String label = labelPredicateObj.asLiteral().getString();
+
+        return NodeFactory.createURI(label);
+    }
+
+    public NodeMapping getNodeMapping(final Model lpg2Rdf, final Resource lpg2rdfConfig){
+        final NodeMapping nodeMapping;
+        if(!lpg2rdfConfig.hasProperty(LPG2RDF.nodeMapping)){
+            throw new IllegalArgumentException("nodeMapping is required!");
+        }
+
+        final Resource nodeMappingResource = lpg2rdfConfig.getProperty(LPG2RDF.nodeMapping).getResource();
+        final RDFNode nodeMappingResourceType = lpg2Rdf.getRequiredProperty(nodeMappingResource, RDF.type).getObject();
+
+        if ( nodeMappingResourceType.equals(LPG2RDF.IRIBasedNodeMapping) ) {
+            final StmtIterator prefixOfIRIsIterator = nodeMappingResource.listProperties(LPG2RDF.prefixOfIRIs);
+            if(!prefixOfIRIsIterator.hasNext()){
+                throw new IllegalArgumentException("prefixOfIRIs is required!");
+            }
+            final RDFNode prefixOfIRIObj = prefixOfIRIsIterator.next().getObject();
+            if(prefixOfIRIsIterator.hasNext()){
+                throw new IllegalArgumentException("More than one prefixOfURI!");
+            }
+
+            final String prefixOfIRIUrl = prefixOfIRIObj.asLiteral().getString();
+            nodeMapping = new NodeMappingToURIsImpl(prefixOfIRIUrl);
```

Comment:
You can directly do the following here.
```suggestion
            return new NodeMappingToURIsImpl(prefixOfIRIUrl);
```